src/serlio/materials/MaterialUtils.cpp

```diff
@@ -168,15 +181,18 @@ MaterialCache getMaterialCache() {
 			continue;
 
 		iterator->setPositionByMemberName(PRT_MATERIALINFO_MAP_VALUE.c_str());
-		MString uuid = iterator->str(0).c_str();
+		uint8_t* uuidPtr = iterator->asUInt8();
+		if (uuidPtr == nullptr)
+			continue;
+		MUuid uuid(uuidPtr);
 
 		existingMaterialInfos.emplace(*hashPtr, uuid);
```

Comment:
you could potentially avoid a MUuid copy by passing uuidPtr directly to .emplace

Reply:
Directly passing uuidPtr to .emplace now